make_prg/subcommands/from_msa.py

```diff
@@ -32,115 +45,110 @@ def register_parser(subparsers):
+    else:
+        all_files = [
+            path.resolve() for path in input_path.iterdir() if path.is_file()
+        ]
+    return all_files
 
-    # Set up file logging
-    formatter = logging.Formatter(
-        fmt="%(levelname)s %(asctime)s %(message)s", datefmt="%d/%m/%Y %I:%M:%S"
-    )
-    handler = (
-        logging.FileHandler(options.log) if options.log else logging.StreamHandler()
-    )
-    handler.setFormatter(formatter)
-    logging.getLogger().addHandler(handler)
 
-    logging.info(
-        "Input parameters max_nesting: %d, min_match_length: %d",
-        options.max_nesting,
-        options.min_match_length,
-    )
+def process_MSA(msa_filepath: Path):
+    global options
+    logger.info(f"Generating PRG for {msa_filepath}...")
+    msa_name = msa_filepath.name
+    locus_name = io_utils.remove_known_input_extensions(msa_filepath).name
 
-    prg_fname = ofile_prefix.with_suffix(".prg")
-    if prg_fname.exists() and options.no_overwrite:
-        logging.info(f"Re-using existing prg file {prg_fname}")
-        aseq = prg_builder.PrgBuilder(
-            options.MSA,
-            alignment_format=options.alignment_format,
-            max_nesting=options.max_nesting,
-            min_match_length=options.min_match_length,
-            prg_file=prg_fname,
-        )
-    else:
-        aseq = prg_builder.PrgBuilder(
-            options.MSA,
+    temp_dir = io_utils.get_temp_dir_for_multiprocess(options.temp_dir)
+    prefix = str(temp_dir / locus_name)
+
+    try:
+        builder = prg_builder.PrgBuilder(
+            locus_name=locus_name,
+            msa_file=msa_filepath,
             alignment_format=options.alignment_format,
             max_nesting=options.max_nesting,
-            min_match_length=options.min_match_length,
+            min_match_length=options.min_match_length
         )
-        logging.info(f"Write PRG file to {prg_fname}")
-        io_utils.write_prg(prg_fname, aseq.prg, options)
-        m = aseq.max_nesting_level_reached
-        logging.info(f"Max_nesting_reached\t{m}")
 
-    if options.output_type.gfa:
-        gfa_fname = ofile_prefix.with_suffix(".gfa")
-        logging.info(f"Write GFA file to {gfa_fname}")
-        io_utils.write_gfa(gfa_fname, aseq.prg)
+        logger.info(f"Writing output files of locus {msa_name}")
+        prg = builder.build_prg()
+
+        if options.output_type.prg or options.output_type.binary:
+            builder.write_prg(prefix, prg)
+
+        if options.output_type.gfa:
+            gfa.GFA_Output.write_gfa(prefix, prg)
+
+        if options.output_type.prg:
+            builder.serialize(f"{prefix}.pickle")
+
+        if options.output_graphs:
+            builder.output_debug_graphs(Path(options.output_prefix + "_debug_graphs"))
+
+    except ValueError as value_error:
+        if "No records found in handle" in value_error.args[0]:
+            logger.warning(f"No records found in MSA {msa_filepath}, skipping...")
```

Comment:
I'd probably agree with this. g

Reply:
I added this just as warning because when running `make_prg` on 37k loci, there were a couple of MSAs that were empty. Or maybe it was another dataset with 10k loci, don't remember exactly... For a single MSA, I do agree erroring out is much better, but I added this as warning thinking on running `make_prg` on a few thousand loci. Do you think that also in this use case, we should error out?

Reply:
Yes

Reply:
Ok, this is implemented with an integration test to make sure things are working, but I can't push to this PR right now. Just have mobile internet connection, and my provider does not allow to share internet with laptop. Will do it asap.

Reply:
Done in https://github.com/iqbal-lab-org/make_prg/pull/35/commits/3891a60a5af486caebffcd7ac0d3ae7d47792a26 . An integration test was also added, but is not part of this PR, will be described in a future PR.